addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/generic/ChannelStateTransformation.java

```diff
@@ -12,76 +12,19 @@
  */
 package org.openhab.binding.mqtt.generic.internal.generic;
 
-import java.lang.ref.WeakReference;
-
 import org.eclipse.jdt.annotation.NonNullByDefault;
-import org.eclipse.jdt.annotation.Nullable;
-import org.eclipse.smarthome.core.transform.TransformationException;
-import org.eclipse.smarthome.core.transform.TransformationService;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * A transformation for a {@link ChannelState}. It is applied for each received value on an MQTT topic.
  *
  * @author David Graeff - Initial contribution
  */
 @NonNullByDefault
-public class ChannelStateTransformation {
-    private final Logger logger = LoggerFactory.getLogger(ChannelStateTransformation.class);
-    private final TransformationServiceProvider provider;
-    private WeakReference<@Nullable TransformationService> transformationService = new WeakReference<>(null);
-    final String pattern;
-    final String serviceName;
-
-    /**
-     * Creates a new MQTT topic subscriber.
-     *
-     * @param stateTopic The MQTT state topic
-     * @param commandTopic The MQTT command topic
-     * @param pattern A transformation pattern, starting with the transformation service
-     *            name,followed by a colon and the transformation itself. An Example:
-     *            JSONPATH:$.device.status.temperature for a json {device: {status: {
-     *            temperature: 23.2 }}}.
-     * @param channelUID The channel UID, used in processMessage() to inform the respective channel
-     *            of an update.
-     * @param cachedValue A value object
-     * @param provider The transformation service provider
-     */
-    public ChannelStateTransformation(String pattern, TransformationServiceProvider provider) {
-        this.provider = provider;
-        int index = pattern.indexOf(':');
-        if (index == -1) {
-            throw new IllegalArgumentException(
-                    "The transformation pattern must consist of the type and the pattern separated by a colon");
-        }
-        String type = pattern.substring(0, index).toUpperCase();
-        this.pattern = pattern.substring(index + 1);
-        this.serviceName = type;
-    }
+public interface ChannelStateTransformation {
```

Comment:
That interface now looks like the standard openHAB transformation interface with a signature of String -> String ;)